### PR TITLE
Csa page redesign

### PR DESCRIPTION
**Pull request by ScottAP108**
The diff is pretty enormous. I advise ignoring almost everything that doesn't have to do with Home, Header, Csa, and CustomContent. A lot of deleted files, a lot of view and template updates that are just switches to add compatibility with the newer framework.

### CODE REVIEW

File: client/js/router.js [hunk 1]
```diff
@@ -59,7 +59,8 @@ module.exports = new (
                 Object.keys( this.views ).forEach( view => this.views[ view ].hide() )
 
                 if( this.views[ resource ] ) this.views[ resource ].show()
-                else this.views[ resource ] = resource === "admin-plus" || resource === 'home'
+
```
Comment (cbaron) [inline client/js/router.js]: Let's revisit the router at the end.  I have versioned package visions

File: client/js/router.js [hunk 2]
```diff
@@ -68,10 +69,7 @@ module.exports = new (
                     : new ( this.resources[ resource ].view )( Object.assign( { factory: this.ViewFactory }, this.resources[ resource ].options ) )
                         .on( 'navigate', data => this.navigate( data.location, data.options ) )
                         
-                if( !/admin/.test( resource ) ) {  
-                    document.body.scrollTop = 0
-                    this.footer.size()
-                }
+                if( !/admin/.test( resource ) ) document.body.scrollTop = 0
```
Comment (cbaron) [inline client/js/router.js]: Do you think the smoothscroll would look better?  Or is is jarring?

File: client/js/views/Csa.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    onSignupBtnClick() { this.emit( 'navigate', 'sign-up' ) },
+
+    postRender() {
+        require('./util/CustomContent').postRender.call(this)
+
+        if( window.location.hash ) {
+            this.els[ this.hashToElement[ window.location.hash.slice(1) ] ].scrollIntoView( { behavior: 'smooth' } )
+        }
+
+        this.Xhr( { method: 'get', resource: 'currentShare' } )
```
Comment (cbaron) [inline client/js/views/Csa.js]: Please put this in a model.  We need to keep tightening up and get to elegant patterns between views and models in an effort to publish this thing eventually.

File: client/js/views/GetInvolved.js [hunk 4]
```diff
@@ -1,29 +1,25 @@
-var CustomContent = require('./util/CustomContent'),
-    GetInvolved = function() { return CustomContent.apply( this, arguments ) }
+module.exports = Object.assign( {}, require('./__proto__'), require('./util/CustomContent'), {
```
Comment (cbaron) [inline client/js/views/GetInvolved.js]: While its nice to not have variables at the top of every file, I prefer putting a required library into a variable for speed.  We should discuss more elegant options though.  Also, I'd like to move towards using classes as well.

You don't have to go back and change all of this though.

File: client/js/views/Home.js [hunk 5]
```diff
@@ -4,6 +4,12 @@ module.exports = Object.assign( {}, require('./__proto__'), {
         joinBtn: 'click'
     },
 
-    onJoinBtnClick() { this.emit( 'navigate', 'sign-up' ) }
+    onJoinBtnClick() { this.emit( 'navigate', 'sign-up' ) },
+
+    postRender() {
+        this.on( 'imgLoaded', () => this.els.container.classList.add('img-loaded') )
```
Comment (cbaron) [inline client/js/views/Home.js]: Are we using GoogleCloudStorage for Patchwork yet ?

File: client/js/router.js (hunk 2 above)
Comment (ScottAP108) [reply client/js/router.js]: I just tried it, and yeah, it's pretty jarring. Feels better to go instantly to the top.

**Pull request merged**